### PR TITLE
Mint specific nft

### PR DESCRIPTION
**Pull request by MGoldstein18**

### CODE REVIEW

**Comment by MGoldstein18** [comment]
@nachoiacovino

**Comment by nachoiacovino** [comment]
you should add me as a reviewer on the right sidebar not tag me :D @MGoldstein18

File: portal/guides/mint-specific-nft-with-nextjs.mdx [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+        <button onClick={connectWithMetamask}>Connect Metamask Wallet</button>
+      )}
+    </div>
+  );
+};
+
+export default Home;
+```
+
+### Create `nftCard` component
```
Comment (nachoiacovino) [inline portal/guides/mint-specific-nft-with-nextjs.mdx]: delete the backticks from here, this is what breaking the build

File: portal/guides/mint-specific-nft-with-nextjs.mdx [hunk 2]
```diff
@@ -0,0 +1,522 @@
+---
+title: Mint a Specific NFT with Next.js
```
Comment (nachoiacovino) [inline portal/guides/mint-specific-nft-with-nextjs.mdx:2]:
```suggestion
title: Mint an specific NFT with Next.js
```

File: portal/guides/mint-specific-nft-with-nextjs.mdx [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+authorAvatar: "/assets/portal/authors/mordi.jpg"
+authorTwitter: "https://twitter.com/mordigoldstein"
+draft: false
+tags: [nft-collection, typescript, sdk, v2]
+summary: "Allow users to select which 1-of-1 NFT to mint"
+image: "/assets/portal/guides/mint-specific-nft-with-nextjs/thumbnail.png"
+date: "2022-04-13"
+updated: "2022-04-13"
+---
+# Mint a Specific NFT
```
Comment (nachoiacovino) [inline portal/guides/mint-specific-nft-with-nextjs.mdx:13]: never have single #
```suggestion

```

File: portal/guides/mint-specific-nft-with-nextjs.mdx [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+
+Allow users to choose which 1-of-1 NFT to mint
```
Comment (nachoiacovino) [inline portal/guides/mint-specific-nft-with-nextjs.mdx:15]:
```suggestion

```

repeated

File: portal/guides/mint-specific-nft-with-nextjs.mdx [hunk 5]
```diff
@@ -0,0 +1,522 @@
+---
+title: Mint a Specific NFT with Next.js
+author: "Mordi Goldstein"
+authorAvatar: "/assets/portal/authors/mordi.jpg"
+authorTwitter: "https://twitter.com/mordigoldstein"
+draft: false
+tags: [nft-collection, typescript, sdk, v2]
+summary: "Allow users to select which 1-of-1 NFT to mint"
```
Comment (nachoiacovino) [inline portal/guides/mint-specific-nft-with-nextjs.mdx:8]:
```suggestion
summary: "Allow users to select which 1-of-1 NFT to mint"
```

We need a better summary here, around 150 words, explaining what the guide is doing

File: portal/guides/mint-specific-nft-with-nextjs.mdx [hunk 6]
```diff
@@ -0,0 +1,522 @@
...
+date: "2022-04-13"
+updated: "2022-04-13"
+---
+# Mint a Specific NFT
+
+Allow users to choose which 1-of-1 NFT to mint
+
+## Introduction
+
+In this guide, we are going to use Next.js to create an NFT collection, build the frontend for users to view the collection, and allow users to select an NFT for minting. We will use the thirdweb [NFT Collection contract](https://portal.thirdweb.com/contracts/nft-collection) which allows you to mint 1 of 1 NFTs.
```
Comment (nachoiacovino) [inline portal/guides/mint-specific-nft-with-nextjs.mdx:19]:
```suggestion
In this guide, we are going to use Next.js to create an NFT collection, build the frontend for users to view the collection, and allow users to select an NFT for minting. We will use the thirdweb [NFT Collection contract](/contracts/nft-collection) which allows you to mint 1 of 1 NFTs. 
```

relative path for internal links

File: portal/guides/mint-specific-nft-with-nextjs.mdx [hunk 7]
```diff
@@ -0,0 +1,522 @@
...
+
+```bash
+npm i @chakra-ui/react @emotion/react@^11 @emotion/styled@^11 framer-motion@^6
+# or
+yarn add @chakra-ui/react @emotion/react@^11 @emotion/styled@^11 framer-motion@^6
+```
+
+Don’t forget to add the `ChakraProvider`, afterwhich your `_app.tsx` should look like this:
+
+```tsx
```
Comment (nachoiacovino) [inline portal/guides/mint-specific-nft-with-nextjs.mdx:70]:
```suggestion
```tsx:_app.tsx
```

add file name to code snippets

File: portal/guides/mint-specific-nft-with-nextjs.mdx [hunk 8]
```diff
@@ -0,0 +1,522 @@
...
+    </div>
+  );
+};
+
+export default Home;
+```
+
+### Create nftCard component
+
+We will display each of our NFTs using an `nftCard` component.  Chakra’s `SimpleGrid`, `Flex` , `Box` and `Image` components will help us build each NFT card. You can create a new file called `nftCard.tsx` in a new folder called `components` .
```
Comment (nachoiacovino) [inline portal/guides/mint-specific-nft-with-nextjs.mdx:137]:
```suggestion
We will display each of our NFTs using an `nftCard` component.  Chakra’s `SimpleGrid`, `Flex` , `Box` and `Image` components will help us build each NFT card. You can create a new file called `NftCard.tsx` in a new folder called `components` .
```

NftCard should be PascalCase not camelCase, components should be capitalized in the file name. Update this throughout the guide and the example code

File: portal/guides/mint-specific-nft-with-nextjs.mdx [hunk 9]
```diff
@@ -0,0 +1,522 @@
...
+
+```tsx
+import { useAddress, useNFTCollection, useMetamask } from '@thirdweb-dev/react';
+```
+
+Implement:
+
+```tsx
+// You can find your contract address in your dashboard after you have created an NFT Collection contract
+  const nftCollectionContractAddress = '<CONTRACT_ADDRESS>';
```
Comment (nachoiacovino) [inline portal/guides/mint-specific-nft-with-nextjs.mdx:269]:
```suggestion
  const nftCollectionAddress = '<CONTRACT_ADDRESS>';
```

too long, we can get rid of contract here

File: portal/guides/mint-specific-nft-with-nextjs.mdx [hunk 10]
```diff
@@ -0,0 +1,522 @@
...
+  const endTime = new Date(Date.now() + 180 * 1000);
+
+  // Set up the NFT metadata
+  const metadata = {
+    metadata: {
+      name,
+      description,
+      image: url,
+    },
+		price,
```
Comment (nachoiacovino) [inline portal/guides/mint-specific-nft-with-nextjs.mdx:332]:
```suggestion
		price,
```
formatting lost here

File: portal/guides/mint-specific-nft-with-nextjs.mdx [hunk 11]
```diff
@@ -0,0 +1,522 @@
...
+    mintStartTime: startTime,
+    mintEndTime: endTime,
+    to: address,
+  };
+
+  // Call the generate method and then call then mint method
+  const response = await nftCollection?.signature
```
Comment (nachoiacovino) [inline portal/guides/mint-specific-nft-with-nextjs.mdx:339]: can we use try/catch async/await here?

**Review by nachoiacovino — CHANGES_REQUESTED** [review]
How do you know no one is sending a POST request with whatever metadata instead of the one you want? 

The selection should be made in the frontend (which one, by tokenId probably) but the metadata should come from the backend, otherwise it's not secure and can be exploited.

**Pull request closed without merging**